core/controllers/acl_decorators_test.py

```diff
@@ -173,133 +378,378 @@ def setUp(self):
+            response = self.get_json(
+                '/mock_play_collection/%s' % self.private_col_id)
+        self.assertEqual(response['collection_id'], self.private_col_id)
+        self.logout()
+
+    def test_logged_in_user_cannot_access_not_owned_private_collection(self):
+        self.login(self.user_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            self.get_json(
+                '/mock_play_collection/%s' % self.private_col_id,
+                expected_status_int=404)
+        self.logout()
+
+    def test_cannot_access_collection_with_invalid_collection_id(self):
+        self.login(self.OWNER_EMAIL)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            self.get_json(
+                '/mock_play_collection/invalid_collection_id',
+                expected_status_int=404)
+        self.logout()
+
+
+class EditCollectionDecoratorTests(test_utils.GenericTestBase):
+    """Tests for can_edit_collection decorator."""
+
+    user_email = 'user@example.com'
+    username = 'user'
+    published_col_id = 'col_id_1'
+    private_col_id = 'col_id_2'
+
+    class MockHandler(base.BaseHandler):
+        GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+        URL_PATH_ARGS_SCHEMAS = {
+            'collection_id': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            }
+        }
+        HANDLER_ARGS_SCHEMAS = {'GET': {}}
+
+        @acl_decorators.can_edit_collection
+        def get(self, collection_id):
+            return self.render_json({'collection_id': collection_id})
+
+    def setUp(self):
+        super(EditCollectionDecoratorTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
+        self.signup(self.MODERATOR_EMAIL, self.MODERATOR_USERNAME)
+        self.signup(self.user_email, self.username)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.set_curriculum_admins([self.CURRICULUM_ADMIN_USERNAME])
+        self.set_moderators([self.MODERATOR_USERNAME])
+        self.set_collection_editors([self.OWNER_USERNAME])
+        self.owner = user_services.get_user_actions_info(self.owner_id)
+        self.mock_testapp = webtest.TestApp(webapp2.WSGIApplication(
+            [webapp2.Route(
+                '/mock_edit_collection/<collection_id>', self.MockHandler)],
+            debug=feconf.DEBUG,
+        ))
+        self.save_new_valid_collection(
+            self.published_col_id, self.owner_id,
+            exploration_id=self.published_col_id)
+        self.save_new_valid_collection(
+            self.private_col_id, self.owner_id,
+            exploration_id=self.private_col_id)
+        rights_manager.publish_collection(self.owner, self.published_col_id)
+
+    def test_can_not_edit_collection_with_invalid_collection_id(self):
```

Comment:
can_not --> cannot

Reply:
replaced here and everywhere else. 

PS: Actually, there were 22 places where `_can_not_` was written instead of `_cannot_` even before this PR. So, I replaced all of them.